internal/statestore/backfill.go

```diff
@@ -134,3 +139,70 @@ func (rb *redisBackend) UpdateBackfill(ctx context.Context, backfill *pb.Backfil
 
 	return nil
 }
+
+// AcknowledgeBackfill - store Backfill's last acknowledgement time
+func (rb *redisBackend) AcknowledgeBackfill(ctx context.Context, id string) error {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		return status.Errorf(codes.Unavailable, "AcknowledgeBackfill, id: %s, failed to connect to redis: %v", id, err)
+	}
+	defer handleConnectionClose(&redisConn)
+
+	currentTime := time.Now().UnixNano()
+	cmds := make([]interface{}, 0)
+	cmds = append(cmds, backfillLastAckTime, currentTime, id)
+
+	_, err = redisConn.Do("ZADD", cmds...)
+	if err != nil {
+		err = errors.Wrap(err, "failed to store backfill's last acknowledgement time")
+		return status.Error(codes.Internal, err.Error())
+	}
+
+	return nil
+
+}
+
+// GetExpiredBackfillIDs - get all backfill IDs which are expired
+func (rb *redisBackend) GetExpiredBackfillIDs(ctx context.Context) ([]string, error) {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Unavailable, "GetExpiredBackfillIDs, failed to connect to redis: %v", err)
+	}
+	defer handleConnectionClose(&redisConn)
+
+	// the same TTL is used for both Backfills and pendingRelease Tickets
+	ttl := rb.cfg.GetDuration("pendingReleaseTimeout")
+	curTime := time.Now()
+	endTimeInt := curTime.Add(-ttl).UnixNano()
+	startTimeInt := 0
+
+	// Filter out tickets that are fetched but not assigned within TTL time (ms).
+	expiredBackfillIds, err := redis.Strings(redisConn.Do("ZRANGEBYSCORE", backfillLastAckTime, startTimeInt, endTimeInt))
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error getting expired backfills %v", err)
+	}
+
+	return expiredBackfillIds, nil
+}
+
+// DeleteExpiredBackfillIDs - delete expired BackfillIDs from a sorted set
+func (rb *redisBackend) DeleteExpiredBackfillIDs(ctx context.Context, backfillIDs []string) error {
```

Comment:
It's not clear to me how this will be used?

Reply:
It would be used by a cleanup service:
1st step `backfillIds, err := GetExpiredBackfills()`
2nd - Return associated tickets for all Bakcfills back to the pool.
Some AcknowledgeBackfill() could occur in between 
3rd step `DeleteExpiredBackfillIDs(backfillIds)`, note that there could be more expired Backfills but we only delete thos e processed.

Reply:
Why not just use the same mechanics as the delete method?  Reduces implementation complexity and possible bugs.

Reply:
Some DeleteBackfill() request can have some issues, others no, so Backend have an opportunity to select on its own.

Reply:
Got an idea now. This can be done as part of `DeleteBackfill`